### ISSUE TITLE
Update physical address error

### ISSUE THREAD

**Issue by armills** [konikvranik/home-assistant#37]
```
ERROR (Thread-21) [asyncio] Exception in callback <bound method HDMINetwork._async_callback of <pycec.network.HDMINetwork object at 0x7fb4e8261400>>
handle: <Handle HDMINetwork._async_callback>
Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 49, in uvloop.loop.Handle._run (uvloop/loop.c:48380)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 436, in _async_callback
    updated = self.get_device(command.src).update_callback(command)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 195, in update_callback
    getattr(self, UPDATEABLE[prop])(command)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 217, in _update_physical_address
    self._type = command.att[2]
IndexError: list index out of range
```

I'm getting the following error in my instance running off of dev. I can load more debugging code if you'd like.

**Comment by konikvranik** [comment]
@armills: Strange. Response to physical address shoulc contain physical address in first two bytes and type in the third one. Can you acquire exact CEC commands by setting debud level to pycec or by observing HDMI network from cec-client?

**Comment by armills** [comment]
OK, included are the full logs below. Based on your description, I think the following message is the problem, which would mean my TV isn't conforming to the standard.

I think the best solution would be to use voluptuous to validate incoming messages.

```
17-01-22 18:32:09 DEBUG (Dummy-23) [pycec] >> 0f:84:00:00
```

```
17-01-22 18:32:04 DEBUG (Dummy-23) [pycec] >> 0f:87:00:e0:91
17-01-22 18:32:05 DEBUG (Thread-21) [pycec] Init done
17-01-22 18:32:05 DEBUG (Thread-21) [pycec] Scanning...
17-01-22 18:32:05 INFO (Thread-21) [pycec] Looking for new devices...
17-01-22 18:32:05 DEBUG (Thread-21) [pycec] Sleep...
17-01-22 18:32:07 DEBUG (Thread-21) [pycec] Found device 0
17-01-22 18:32:07 DEBUG (Dummy-23) [pycec] >> 02:89:01
17-01-22 18:32:07 DEBUG (Thread-21) [pycec] Found device 1
17-01-22 18:32:08 DEBUG (Thread-21) [pycec] Starting device 0
17-01-22 18:32:08 DEBUG (Thread-21) [pycec] << f0:83
17-01-22 18:32:08 DEBUG (Thread-21) [pycec] << f0:46
17-01-22 18:32:09 INFO (MainThread) [homeassistant.core] Bus:Handling <Event platform_discovered[L]: platform=hdmi_cec, service=load_platform.switch, discovered=new=['hdmi_cec.hdmi_0']>
17-01-22 18:32:09 DEBUG (Dummy-23) [pycec] >> 02:89:01
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f0:71
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f0:1a
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f0:8c
17-01-22 18:32:09 DEBUG (Dummy-23) [pycec] >> 02:89:04:00
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f0:8f
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] Starting device 1
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:83
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:46
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:71
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:1a
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:8c
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f1:8f
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] Found device 5
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] Found device 8
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] Starting device 5
17-01-22 18:32:09 DEBUG (Dummy-23) [pycec] >> 0f:84:00:00
17-01-22 18:32:09 INFO (MainThread) [homeassistant.loader] Loaded switch.hdmi_cec from homeassistant.components.switch.hdmi_cec
17-01-22 18:32:09 DEBUG (Thread-21) [pycec] << f5:83
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f5:46
17-01-22 18:32:10 INFO (MainThread) [homeassistant.core] Bus:Handling <Event platform_discovered[L]: platform=hdmi_cec, service=load_platform.switch, discovered=new=['hdmi_cec.hdmi_1']>
17-01-22 18:32:10 DEBUG (Dummy-23) [pycec] >> 02:90:00
MutexImpl::Unlock error 11 (1)
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f5:71
17-01-22 18:32:10 INFO (MainThread) [homeassistant.core] Bus:Handling <Event platform_discovered[L]: platform=hdmi_cec, service=load_platform.switch, discovered=new=['hdmi_cec.hdmi_5']>
17-01-22 18:32:10 DEBUG (Dummy-23) [pycec] >> 1f:84:10:00:01
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f5:1a
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f5:8c
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f5:8f
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] Starting device 8
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:83
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:46
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:71
17-01-22 18:32:10 INFO (MainThread) [homeassistant.components.switch] Setting up switch.hdmi_cec
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:1a
17-01-22 18:32:10 DEBUG (Dummy-23) [pycec] >> 12:47:58:42:4d:43
17-01-22 18:32:10 INFO (Thread-11) [homeassistant.components.switch.hdmi_cec] Setting up HDMI devices ['hdmi_cec.hdmi_0']
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:8c
17-01-22 18:32:10 INFO (MainThread) [homeassistant.core] Bus:Handling <Event platform_discovered[L]: platform=hdmi_cec, service=load_platform.switch, discovered=new=['hdmi_cec.hdmi_8']>
17-01-22 18:32:10 DEBUG (Thread-21) [pycec] << f8:8f
17-01-22 18:32:10 INFO (MainThread) [homeassistant.components.switch] Setting up switch.hdmi_cec
17-01-22 18:32:10 ERROR (Thread-21) [asyncio] Exception in callback <bound method HDMINetwork._async_callback of <pycec.network.HDMINetwork object at 0x7f24f1161898>>
handle: <Handle HDMINetwork._async_callback>
Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 49, in uvloop.loop.Handle._run (uvloop/loop.c:48380)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 436, in _async_callback
    updated = self.get_device(command.src).update_callback(command)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 195, in update_callback
    getattr(self, UPDATEABLE[prop])(command)
  File "/usr/local/lib/python3.5/site-packages/pycec/network.py", line 217, in _update_physical_address
    self._type = command.att[2]
IndexError: list index out of range
17-01-22 18:32:10 INFO (MainThread) [homeassistant.components.switch] Setting up switch.hdmi_cec
17-01-22 18:32:10 INFO (Thread-5) [homeassistant.components.switch.hdmi_cec] Setting up HDMI devices ['hdmi_cec.hdmi_1']
17-01-22 18:32:11 DEBUG (Dummy-23) [pycec] >> 12:90:01
17-01-22 18:32:11 INFO (Thread-3) [homeassistant.components.switch.hdmi_cec] Setting up HDMI devices ['hdmi_cec.hdmi_5']
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=switch.hdmi_0, new_state=<state switch.hdmi_0=on; vendor_name=Unknown, hidden=True, type=TV, physical_address=0.0.0.0, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 0 () @ 2017-01-22T18:32:11.118121-05:00>>
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=switch.hdmi_1, new_state=<state switch.hdmi_1=unknown; vendor_name=Unknown, hidden=True, type=Recorder, physical_address=1.0.0.0, type_id=1, icon=mdi:microphone, vendor_id=0, friendly_name=Recorder 1 (XBMC) @ 2017-01-22T18:32:11.536523-05:00>>
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=switch.hdmi_5, new_state=<state switch.hdmi_5=unknown; vendor_name=Unknown, hidden=True, type=TV, physical_address=None, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 5 () @ 2017-01-22T18:32:11.716645-05:00>>
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state switch.hdmi_1=unknown; vendor_name=Unknown, hidden=True, type=Recorder, physical_address=1.0.0.0, type_id=1, icon=mdi:microphone, vendor_id=0, friendly_name=Recorder 1 (XBMC) @ 2017-01-22T18:32:11.536523-05:00>, entity_id=switch.hdmi_1, new_state=<state switch.hdmi_1=off; vendor_name=Unknown, hidden=True, type=Recorder, physical_address=1.0.0.0, type_id=1, icon=mdi:microphone, vendor_id=0, friendly_name=Recorder 1 (XBMC) @ 2017-01-22T18:32:11.719343-05:00>>
17-01-22 18:32:11 INFO (MainThread) [homeassistant.components.switch] Setting up switch.hdmi_cec
17-01-22 18:32:11 INFO (Thread-11) [homeassistant.components.switch.hdmi_cec] Setting up HDMI devices ['hdmi_cec.hdmi_8']
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=None, entity_id=switch.hdmi_8, new_state=<state switch.hdmi_8=unknown; vendor_name=Unknown, hidden=True, type=TV, physical_address=None, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 8 () @ 2017-01-22T18:32:11.740193-05:00>>
17-01-22 18:32:11 DEBUG (Dummy-23) [pycec] >> 5f:84:30:00:05
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state switch.hdmi_5=unknown; vendor_name=Unknown, hidden=True, type=TV, physical_address=None, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 5 () @ 2017-01-22T18:32:11.716645-05:00>, entity_id=switch.hdmi_5, new_state=<state switch.hdmi_5=on; vendor_name=Unknown, hidden=True, type=Audio, physical_address=3.0.0.0, type_id=5, icon=mdi:speaker, vendor_id=0, friendly_name=Audio 5 () @ 2017-01-22T18:32:11.895649-05:00>>
17-01-22 18:32:11 DEBUG (Dummy-23) [pycec] >> 82:90:01
17-01-22 18:32:11 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: old_state=<state switch.hdmi_8=unknown; vendor_name=Unknown, hidden=True, type=TV, physical_address=None, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 8 () @ 2017-01-22T18:32:11.740193-05:00>, entity_id=switch.hdmi_8, new_state=<state switch.hdmi_8=off; vendor_name=Unknown, hidden=True, type=TV, physical_address=None, type_id=0, icon=mdi:television, vendor_id=0, friendly_name=TV 8 () @ 2017-01-22T18:32:11.998348-05:00>>
```

**Comment by konikvranik** [comment]
@armills: Preparing validation for all the CEC stuff is too expensive.
I start with cheapest solution - to check message length for this particular case and handle also not CEC conforming info.

**Comment by konikvranik** [comment]
fixed in konikvranik/pyCEC#38

**Comment by armills** [comment]
OK, works for me.
